src/components/GlobalPreferences/GlobalPreferences.js

```diff
@@ -0,0 +1,250 @@
+            </main>
+          </React.Fragment>
+        )}
+      </Layout>
+    </div>
+  )
+}
+
+GlobalPreferences.propTypes = {
+  dao: PropTypes.string,
+  helpScoutOptedOut: PropTypes.bool,
+  onClose: PropTypes.func.isRequired,
+  onHelpScoutOptedOutChange: PropTypes.func.isRequired,
+  opened: PropTypes.bool,
+  toast: PropTypes.func,
+  wrapper: AragonType,
+}
+
+function useGlobalPreferences(opened) {
+  const [currentSection, setCurrentSection] = useState(0)
+  const handleNavigation = useCallback(index => {
+    const { hash } = window.location
+    const rest = hash.substr(0, hash.indexOf(GLOBAL_PREFERENCES_QUERY_PARAM))
+    window.location.hash = `${rest}?p=/${PATHS[index]}`
+  }, [])
+
+  useEffect(() => {
+    if (!opened) {
+      return
+    }
+    const { hash } = window.location
+    const path = hash.substr(
+      hash.indexOf(GLOBAL_PREFERENCES_QUERY_PARAM) +
+        GLOBAL_PREFERENCES_QUERY_PARAM.length
+    )
+    setCurrentSection(
+      SECTIONS.has(path) ? PATHS.findIndex(item => item === path) : 0
+    )
+  }, [opened])
+
+  return { setCurrentSection, currentSection, handleNavigation }
+}
+
+function Close({ onClick }) {
+  return (
+    <div
+      css={`
+        text-align: right;
+        padding-top: ${2.5 * GU}px;
+        padding-right: ${3 * GU}px;
+      `}
+    >
+      <ButtonIcon onClick={onClick} label="Close">
+        <IconClose />
+      </ButtonIcon>
+    </div>
+  )
+}
+
+Close.propTypes = {
+  onClick: PropTypes.func.isRequired,
+}
+
+function AnimatedGlobalPreferences({ opened, ...props }) {
+  const { below } = useViewport()
+  const smallView = below('medium')
+  const theme = useTheme()
+
+  return (
+    <Transition
+      native
+      items={opened}
+      from={{ opacity: 0, enterProgress: 0, blocking: false }}
+      enter={{ opacity: 1, enterProgress: 1, blocking: true }}
+      leave={{ opacity: 0, enterProgress: 1, blocking: false }}
+      config={springs.smooth}
+    >
+      {show =>
+        show &&
+        /* eslint-disable react/prop-types */
+        // z-index 2 on mobile keeps the menu above this preferences modal
+        (({ opacity, enterProgress, blocking }) => (
+          <AnimatedWrap
+            accent={theme.accent}
+            surface={theme.surface}
+            style={{
+              zIndex: smallView ? 2 : 5,
+              pointerEvents: blocking ? 'auto' : 'none',
+              opacity,
+              transform: enterProgress.interpolate(
+                v => `
+                  translate3d(0, ${(1 - v) * 10}px, 0)
+                  scale3d(${1 - (1 - v) * 0.03}, ${1 - (1 - v) * 0.03}, 1)
+                `
+              ),
+            }}
+          >
+            <GlobalPreferences {...props} opened={opened} />
+          </AnimatedWrap>
+        ))
+      /* eslint-enable react/prop-types */
+      }
+    </Transition>
+  )
+}
+
+AnimatedGlobalPreferences.propTypes = {
+  opened: PropTypes.bool,
+}
+
+const AnimatedWrap = styled(animated.div)`
```

Comment:
Just as a note: using a separate component for `animated` components is not needed anymore since the latest version of `babel-plugin-styled-components`!

![](https://user-images.githubusercontent.com/36158/60993882-9d669380-a34f-11e9-8c0a-11210ae71fc4.gif)

Reply:
I was getting the issue of `animated.div` is not a tag/component, so let's update when that feature is released :)